### PR TITLE
Restructured clone_package to be more generalizable

### PR DESCRIPTION
**Pull request by maier-m**

### CODE REVIEW

File: R/clone_package.R [hunk 1]
```diff
@@ -1,287 +1,325 @@
-#' Return the object metadata of each data object in a Data Package
+#' Clones objects between DataONE Member Nodes.
 #'
-#' This function returns the formatId, fileName and identifier of each data
-#' object in a Data Package.  This is a helper function for the function
-#' 'clone_package' - which copies a dataOne Data Package from one member node to
-#' another.
+#' @description Clones objects between DataONE member nodes.
```
Comment (isteves) [inline R/clone_package.R]: Add info about tokens/authentication

File: R/clone_package.R [hunk 2]
```diff
@@ -1,287 +1,325 @@
...
-#' @return (list) List of all the identifiers in the new Data Package.
-clone_one_package <- function(resource_map_pid, from, to, public = FALSE) {
-    stopifnot(is.character(resource_map_pid))
-    stopifnot(methods::is(from, "MNode"))
-    stopifnot(methods::is(to, "MNode"))
-    stopifnot(is.logical(public))
+    # Get DataObject
+    data_obj <- dataone::getDataObject(from, pid)
+
+    # Adujst sysmeta
```
Comment (isteves) [inline R/clone_package.R]: adjust*

File: R/clone_package.R [hunk 3]
```diff
@@ -1,287 +1,325 @@
...
+    upload_pid <- tryCatch({
 
-    return(response)
+        dataone::createObject(x = to@mn,
+                              pid = data_obj@sysmeta@identifier,
+                              sysmeta = data_obj@sysmeta,
+                              dataobj = data_obj@data)
+
+    }, error = function(e) {
+        warning(e)
```
Comment (isteves) [inline R/clone_package.R]: why not write a warning message?

File: R/clone_package.R [hunk 4]
```diff
@@ -1,287 +1,325 @@
...
-
-            response[["resource_map"]] <- updated_resource_map_pid
-            response[["child_packages"]] <- unname(child_resource_map_pids)
+                          clone_children = FALSE,
+                          new_pid = TRUE,
+                          new_submitter = NULL,
+                          change_origin_node = FALSE,
+                          change_auth_node = FALSE) {
+
+    if (!(is.character(resource_map_pid) & nchar(resource_map_pid) > 0)) {
```
Comment (isteves) [inline R/clone_package.R]: These function checks could probably be condensed, perhaps by using something like `assertthat`, which does some of the error-message to English translation for you: https://github.com/hadley/assertthat

File: R/clone_package.R (hunk 3 above)
Comment (maier-m) [reply R/clone_package.R]: Turn error into a warning

File: R/clone_package.R (hunk 4 above)
Comment (maier-m) [reply R/clone_package.R]: Good tip, will keep for future reference but don't feel is necessary to go back through these as the changes will be purely aesthetic.

File: R/clone_package.R (hunk 1 above)
Comment (maier-m) [reply R/clone_package.R]: Done

File: R/clone_package.R (hunk 2 above)
Comment (maier-m) [reply R/clone_package.R]: Done

**Pull request merged**